symfit/core/minimizers.py

```diff
@@ -60,11 +60,13 @@ def _baseobjective_from_callable(self, func, objective_type=MinimizeModel):
             if isinstance(func, BaseModel):
                 model = func
             else:
-                # Minimize the provided custom objective instead. This why want to
-                # minimize a CallableNumericalModel, thats what they are for.
-                model = CallableNumericalModel(func,
-                                               params=self.parameters,
-                                               independent_vars=[])
+                # Minimize the provided custom objective instead. We therefore
+                # wrap it into a CallableNumericalModel, thats what they are for
+                y = sympy.Dummy('y')
```

Comment:
How bad is this going to go if the model uses a variable `'y'` somewhere? I guess it should be OK since it's a Dummy, right?
Does it work in general with all the sorting into variables/parameters if users make a model with both Dummies and Variables with identical names? That question is out of scope though.

Reply:
Even `Dummy('y') == Dummy('y')` is `False`, the name is not important, essentially just the memory address.

Good question about the sorting, in principle they do have a name so I guess they would rank as the same for alphabetical ordering. That could actually go horribly wrong when both of those happen to be in the models signature. (Not a problem here, although I will nonetheless remove the name and let sympy generate one for us just in case.)